enumerable_methods.rb

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/ruby
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    for x in self # rubocop:disable Style/For
+      yield(x)
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    for x in self # rubocop:disable Style/For
+      yield(x, i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    new_array = []
+    my_each do |x|
+      new_array.push(x) if yield(x)
+    end
+    new_array
+  end
+
+  def my_all?
+    unless block_given?
+      my_each do |x|
+        return false unless x
+      end
+      return true
+    end
+    my_each do |x|
+      return false unless yield(x)
+    end
+    true
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
Sample case:
```ruby
array = Array.new(100) { rand(0...9)
array.my_all?(Integer).to be array.all?(Integer)
```
Should return `true` if `my_all?` works as expected.